cmd/internal/scdiff/app/compare.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2023 OpenSSF Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package app
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/spf13/cobra"
+
+	"github.com/ossf/scorecard/v4/cmd/internal/scdiff/app/compare"
+	"github.com/ossf/scorecard/v4/cmd/internal/scdiff/app/format"
+	"github.com/ossf/scorecard/v4/pkg"
+)
+
+//nolint:gochecknoinits // common for cobra apps
+func init() {
+	rootCmd.AddCommand(compareCmd)
+}
+
+var (
+	errMissingInputFiles = errors.New("must provide at least two files")
+	errResultsDiffer     = errors.New("results differ")
+	errNumResults        = errors.New("number of results being compared differ")
+
+	compareCmd = &cobra.Command{
+		Use:   "compare [flags] FILE1 FILE2",
+		Short: "Compare Scorecard results",
+		Long:  `Compare Scorecard results`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) < 2 {
+				return errMissingInputFiles
+			}
+			f1, err := os.Open(args[0])
+			if err != nil {
+				return fmt.Errorf("opening %q: %w", args[0], err)
+			}
+			defer f1.Close()
+			f2, err := os.Open(args[1])
+			if err != nil {
+				return fmt.Errorf("opening %q: %w", args[1], err)
+			}
+			defer f2.Close()
+			cmd.SilenceUsage = true  // disables printing Usage
+			cmd.SilenceErrors = true // disables the "Error: <err>" message
+			return compareReaders(f1, f2, os.Stderr)
+		},
+	}
+)
+
+func compareReaders(x, y io.Reader, output io.Writer) error {
+	// results are currently newline delimited
+	xs := bufio.NewScanner(x)
+	ys := bufio.NewScanner(y)
+	for {
+		if shouldContinue, err := advanceScanners(xs, ys); err != nil {
+			return err
+		} else if !shouldContinue {
+			break
+		}
+		xResult, yResult, err := loadResults(xs, ys)
+		if err != nil {
+			return err
+		}
+		if !compare.Results(&xResult, &yResult) {
```

Comment:
We know exactly where in the JSON there's a diff between `r1`, `r2` but the return value of this function is a boolean and then prints the entire contents of the JSON, instead of giving the first location where there's a diff. Trying to understand why we use `cmp.Diff` and then implement `compare.Results` separately?

Reply:
My intention was using our own compare func to ignore certain fields, but then fall back on cmp.Diff to pretty print.

Reply:
Gotcha. I think it'd be cleaner if we committed to a single comparison function that included diff location, because it seems like repeated work to diff twice, and also that `go-cmp` might point out more differences than we care between two scorecard runs, but that's up to whatever seems to you to be more maintainable.